src/v/iceberg/values_json.cc

```diff
@@ -0,0 +1,674 @@
+        auto us_sub_sec = sv_to_us(u_str);
+        auto since_midnight = sv_to_time_point("%T", t_str)
+                              - sv_to_time_point("%T", "00:00:00");
+        return time_value{(since_midnight + us_sub_sec) / 1us};
+    }
+    value operator()(const timestamp_type&) {
+        using namespace std::chrono_literals;
+        auto str = parse_string_view(data_, "timestamp_value");
+        auto split = split_char(str, '.');
+        if (!split.has_value()) {
+            throw std::invalid_argument(fmt::format(
+              "Expected fractional part for timestamp_value, got '{}'", str));
+        }
+        auto [t_str, u_str] = split.value();
+        auto us_sub_sec = sv_to_us(u_str);
+        auto since_epoch = sv_to_time_point("%FT%T", t_str).time_since_epoch();
+        return timestamp_value{(since_epoch + us_sub_sec) / 1us};
+    }
+    value operator()(const timestamptz_type&) {
+        using namespace std::chrono_literals;
+        auto str = parse_string_view(data_, "timestamptz_value");
+        auto split = split_char(str, '.');
+        if (!split.has_value()) {
+            throw std::invalid_argument(fmt::format(
+              "Expected fractional part for timestamptz_value, got '{}'", str));
+        }
+        auto [t_str, rest] = split.value();
+        split = split_char(rest, '+');
+        if (!split.has_value()) {
+            throw std::invalid_argument(fmt::format(
+              "Expected offset part for timestamptz_value, got '{}'", str));
+        }
+        auto [u_str, _] = split.value();
+        auto us_sub_sec = sv_to_us(u_str);
+        auto since_epoch = sv_to_time_point("%FT%T", t_str).time_since_epoch();
+        return timestamptz_value{(since_epoch + us_sub_sec) / 1us};
+    }
+    value operator()(const string_type&) {
+        auto str = parse_string_view(data_, "string_value");
+        return string_value{iobuf::from(str)};
+    }
+    value operator()(const fixed_type& t) {
+        auto str = parse_string_view(data_, "fixed_value");
+        if (t.length * 2 != str.size()) {
+            throw std::invalid_argument(fmt::format(
+              "Expected {} hex digits for {}: got {}",
+              t.length * 2,
+              t,
+              str.size()));
+        }
+        return fixed_value{hex_str_to_iobuf(str)};
+    }
+    value operator()(const uuid_type&) {
+        auto str = parse_string_view(data_, "uuid_value");
+        try {
+            return uuid_value{uuid_t::from_string(str)};
+        } catch (const boost::wrapexcept<std::runtime_error>& e) {
+            throw std::invalid_argument(
+              fmt::format("Failed to parse uuid: {}", e.what()));
+        }
+    }
+    value operator()(const binary_type&) {
+        auto str = parse_string_view(data_, "binary_value");
+        return binary_value{hex_str_to_iobuf(str)};
+    }
+    value operator()(const decimal_type& t) {
+        // TODO(oren): need to support negative scale? see datatypes.h
```

Comment:
See table in https://iceberg.apache.org/spec/#json-single-value-serialization

Reply:
https://github.com/apache/iceberg/blob/b128bba57f613f23ed773f0fa6330c1d2bbf8a39/core/src/test/java/org/apache/iceberg/TestSingleValueParser.java#L53-L55
https://github.com/apache/iceberg/blob/b128bba57f613f23ed773f0fa6330c1d2bbf8a39/core/src/test/java/org/apache/iceberg/TestSingleValueParser.java#L143

Reply:
For rust:

https://github.com/apache/iceberg-rust/blob/ae04c8a790b0d949d1f5303f91a6a6c2c40a9f9e/crates/iceberg/src/spec/values.rs#L1037

https://github.com/paupino/rust-decimal/blob/46fb4c3c517bc0c27cd534b65f9e8b57c24ba18e/tests/decimal_tests.rs#L3272

Reply:
Yeah we need to support scientific notation, but the point here is that up-scaling is signified by a negative value for scale, whereas unsigned int is baked into decimal_value currently. We should change that, but I left it off initially to jeep this commit PR fully isolated.

Reply:
i think it's fine for a follow up since we don't currently support it at the `iceberg::datatypes` level 

https://redpandadata.atlassian.net/browse/CORE-8835

Reply:
sounds good